src/us/kbase/assemblyhomology/filters/KBaseAuthenticatedFilterFactory.java

```diff
@@ -0,0 +1,191 @@
+	 * (one of prod, appdev, next, or ci) the filter is associated with. This allows for
+	 * a single Assembly Homology service to be used for multiple KBase environments, if
+	 * desired. If omitted, the default of 'prod' is used.
+	 * @param config the filter configuration.
+	 * @throws MinHashFilterFactoryInitializationException if an initialization error occurs.
+	 */
+	public KBaseAuthenticatedFilterFactory(final Map<String, String> config)
+			throws MinHashFilterFactoryInitializationException {
+		checkNotNull(config, "config");
+		final String env;
+		if (config.containsKey(CONFIG_ENV)) {
+			env = config.get(CONFIG_ENV);
+			if (!ENVS.contains(env)) {
+				throw new MinHashFilterFactoryInitializationException(
+						"Illegal KBase filter environment value: " + env);
+			}
+		} else {
+			env = ENVS.get(0);
+		}
+		try {
+			id = new FilterID("kbase" + env);
+		} catch (MissingParameterException | IllegalParameterException e) {
+			throw new RuntimeException("This should be impossible", e);
+		}
+		url = getURL(config);
+		if (!url.getProtocol().equals("https")) {
+			insecure = true;
+			// logging is checked manually. Check that logging occurs if you make changes here
+			LoggerFactory.getLogger(getClass()).info(String.format(
+					"Workspace url %s is insecure. It is strongly recommended to use https.",
+					url));
+		} else {
+			// can't really test this easily
+			insecure = false;
+		}
+	}
+
+	private URL getURL(final Map<String, String> config)
+			throws MinHashFilterFactoryInitializationException {
+		final String surl = config.get(CONFIG_URL);
+		if (isNullOrEmpty(surl)) {
+			throw new MinHashFilterFactoryInitializationException(
+					"KBase filter requires key 'url' in config");
+		}
+		final URL u;
+		try {
+			u = new URL(surl);
+		} catch (MalformedURLException e) {
+			throw new MinHashFilterFactoryInitializationException(
+					"KBase filter url malformed: " + surl);
+		}
+		final String ver;
+		try {
+			final WorkspaceClient cli = new WorkspaceClient(u);
+			cli.setIsInsecureHttpConnectionAllowed(true); // ok since no token
+			ver = cli.ver();
+		} catch (JsonClientException | IOException e) {
+			// not sure how a jsonclient exception could actually happen here
+			throw new MinHashFilterFactoryInitializationException(String.format(
+					"KBase filter failed contacting workspace at url %s: %s",
+					u, e.getMessage()), e);
+		}
+		if (Version.valueOf(ver).lessThan(Version.valueOf("0.8.0"))) {
+			// this is annoying to test. Just test manually by bumping the version above
+			// and looking for appropriate test failures
+			throw new MinHashFilterFactoryInitializationException(
+					"KBase filter requires workspace version >= 0.8.0, was " + ver);
+		}
+		return u;
+	}
+	
+	@Override
+	public FilterID getID() {
+		return id;
+	}
+
+	@Override
+	public Optional<String> getAuthSource() {
+		return Optional.of(id.getName());
+	}
+
+	@Override
+	public KBaseAuthenticatedFilter getFilter(
```

Comment:
This function is confusing to me, does it return a filter that just checks if you are authenticated, or it returns a filter with data that has been filtered upon authentication

Reply:
Ok I reread the comment at the top